src/views/ResourceViewer.vue

```diff
@@ -255,233 +255,227 @@
-					return 0
-        }
-
-				const
-						channels = this.tree.details.hasVoice
+    isAudioLibraryEmpty() {
+      return !this.$store.state.trees.ui.audioFiles.length
+    },
+
+    totalPiecesOfBlockContent() {
+      return this.blocks.length * lodash.size(this.enabledLanguages)
+    },
+
+    percentagePiecesWithContent() {
+      if (!this.blocks.length) {
+        return 0
+      }
+
+      const
+        channels = this.tree.details.hasVoice
 								+ this.tree.details.hasSms
 								+ this.tree.details.hasUssd
-								+ 1, // + 1 = reviewed
-						completed = this.totalPiecesOfBlockContentWithVoice
+								+ 1 // + 1 = reviewed
+      const completed = this.totalPiecesOfBlockContentWithVoice
 								+ this.totalPiecesOfBlockContenWithSms
 								+ this.totalPiecesOfBlockContenWithUssd
-								+ this.totalPiecesOfBlockContentReviewed,
-						total = (this.totalPiecesOfBlockContent * channels)
-
-				return ((completed / total) * 100).toFixed(0)
-			},
-
-			totalPiecesOfBlockContentWithVoice() {
-				if (!this.tree.details.hasVoice) {
-					return 0
-				}
-
-				return lodash.reduce(this.enabledLanguages, (sum, langId) => {
-					return lodash.reduce(this.blocks, (sum, block) => sum + +!!block.audioFiles[langId], sum)
-				}, 0)
-			},
-
-			totalPiecesOfBlockContenWithSms() {
-				if (!this.tree.details.hasSms) {
-					return 0
-				}
-
-				return lodash.reduce(this.enabledLanguages, (sum, langId) => {
-					return lodash.reduce(this.blocks, (sum, block) => sum + +!!block.smsContent[langId], sum)
-				}, 0)
-			},
-
-			totalPiecesOfBlockContenWithUssd() {
-				if (!this.tree.details.hasUssd) {
-					return 0
-				}
-
-				return lodash.reduce(this.enabledLanguages, (sum, langId) => {
-					return lodash.reduce(this.blocks, (sum, block) => sum + +!!block.ussdContent[langId], sum)
-				}, 0)
-			},
-
-			totalPiecesOfBlockContentReviewed() {
-				return lodash.reduce(this.enabledLanguages, (sum, langId) => {
-					// using lodash to fetch this one because reviewed hash may be absent
-					return lodash.reduce(this.blocks, (sum, block) => sum + +!!lodash.get(block, `customData.reviewed.${langId}`), sum)
-				}, 0)
-			}
-		},
-
-		methods: {
-      ...mapActions(['initializeTreeModel']),
+								+ this.totalPiecesOfBlockContentReviewed
+      const total = (this.totalPiecesOfBlockContent * channels)
+
+      return ((completed / total) * 100).toFixed(0)
+    },
+
+    totalPiecesOfBlockContentWithVoice() {
+      if (!this.tree.details.hasVoice) {
+        return 0
+      }
+
+      return lodash.reduce(this.enabledLanguages, (sum, langId) => lodash.reduce(this.blocks, (sum, block) => sum + +!!block.audioFiles[langId], sum), 0)
+    },
+
+    totalPiecesOfBlockContenWithSms() {
+      if (!this.tree.details.hasSms) {
+        return 0
+      }
+
+      return lodash.reduce(this.enabledLanguages, (sum, langId) => lodash.reduce(this.blocks, (sum, block) => sum + +!!block.smsContent[langId], sum), 0)
```

Comment:
I get a big complaint about line length here (maybe from my global config?) and I sort of agree with it. Can we add a limit of 100? Would it be a massive pain to go through these?

Reply:
Right now the line limit is 100 characters per line. But these errors have not been fixed yet. We have 1296 errors and 268 warnings found. I have created the VMO-3551 to address this and brainstorm more ESLint changes that we need.

Reply:
Great. Happy to leave these for now in that case and try and avoid adding more!

Reply:
@poojahpatil90 I realise I may have misunderstood. Are you suggesting we delay the line length fixes for later? I'm happy with that in order to push this to master more quickly as long as we create a separate ticket for fixes we are leaving until later. We can keep 3551 to track your work on this first phase and in order to have something to call done. Then we should have a separate ticket for further improvements and make Jon aware of this to prioritise. 

My other concern is that if we don't have a clean master and an automated build fail on linting errors (which I think we discussed adding to travis as part of this task?) we won't keep things clean going forward. Can we relax the line length rule and any other rules we will not make fixes for now in eslintrc so we can get to that passing build? Then we can detail these temporary relaxations in the new ticket.